src/x86/unix64.S

```diff
@@ -334,18 +351,31 @@ E(L(load_table), UNIX64_RET_ST_XMM0_RAX)
 E(L(load_table), UNIX64_RET_ST_RAX_XMM0)
 	movq	8(%rsi), %xmm0
 	jmp	L(l2)
-E(L(load_table), UNIX64_RET_ST_XMM0_XMM1)
+E(L(load_table), UNIX64_RET_ST_XMM0_XMM1_64)
 	movq	8(%rsi), %xmm1
 	jmp	L(l3)
+E(L(load_table), UNIX64_RET_ST_XMM0_XMM1_128)
+	movq	16(%rsi), %xmm1
+	jmp	L(l4)
+E(L(load_table), UNIX64_RET_ST_XMM0)
+	movdqu  (%rsi), %xmm0
+	ret
 E(L(load_table), UNIX64_RET_ST_RAX_RDX)
 	movq	8(%rsi), %rdx
+    jmp    L(l2)
+E(L(load_table), UNIX64_RET_X86_ST0)
+	movdqa  16(%rsi), %xmm1
```

Comment:
I suspect that this implementation may omit the 3rd double element when passing a 3-element array of doubles as an argument. Can you add a suite of tests for passing matrices and vectors as arguments and check whether all of them will pass?